src/components/gamesList.js

```diff
@@ -1,73 +1,172 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
 import axios from "axios";
 // import Modali, { useModali } from "modali";
 import Modal from "react-modal";
 import { DateTime } from "luxon";
 import { Link } from "react-router-dom";
+import GameDetail from "./gameDetail";
 
-
-export default function GamesList({ token }) {
-    const [allGamesList, setAllGamesList] = useState([]);
+export default function GamesList({ token, games, listType, listTitle }) {
     const [modalIsOpen, setModalIsOpen] = useState(false);
-    const [modalData, setModalData] = useState({});
-    const [error, setError] = useState('')
-    const [joinRequestSent, setJoinRequestSent] = useState(false)
-    const [currentGame, setCurrentGame] = useState(null)
+    const [error, setError] = useState(null);
+    const [joinRequestSent, setJoinRequestSent] = useState(false);
+    const [currentGame, setCurrentGame] = useState(null);
 
-    Modal.setAppElement('#root');
+    Modal.setAppElement("#root");
 
     const handleOpenModal = (game) => {
-        console.log("click open");
+        console.log("click modal open");
         console.log(game.id);
+        // console.log(game)
         setModalIsOpen(true);
-        setModalData(game);
-        console.log(modalData);
+        setCurrentGame(game);
+        console.log(currentGame);
+        console.log(modalIsOpen);
+        console.log(listType)
     };
 
-    const handleCloseModal = () => {
+    const handleCloseModal = (game) => {
         console.log("click close");
+        console.log(game);
         setModalIsOpen(false);
     };
 
-    const handleJoinClick = (game) => {
-        console.log("join click")
-        console.log(game)
-        setCurrentGame(game)
-        axios 
-            .post(`https://teammate-app.herokuapp.com/session/${game.id}/guest`, {}, {
+    const handleCancelRequest = (game) => {
+        console.log("click cancel request");
+        console.log(game);
+        axios
+            .delete(
+                `https://teammate-app.herokuapp.com/session/${game.id}/guest/${game.guest_info[0].id}/`,
+                {
+                    headers: {
+                        Authorization: `Token ${token}`,
+                    },
+                }
+            )
+            .catch((res) => {
+                // setError(error.message);
+                alert(res.detail);
+            });
+        // Does FE need to anything else with this? Or does the next person in the queue get updated in BE?
```

Comment:
I don't remember if the user(host) is confirming the join request if it is like that, I guess user-host should get a cancel notification and create another game host (to make it simple)....?

Reply:
Agreed. I don't know at this point if that's something for the front end or the backend. I'll make a card on Trello for it.